more CI problems this time docs

https://github.com/valhalla/valhalla/actions/runs/6554539168/job/17801764898

```
Run mkdocs gh-deploy --force --strict
INFO    -  DeprecationWarning: 'materialx.emoji.twemoji' is deprecated.
Material emoji logic has been officially moved into mkdocs-material
version 9.[4](https://github.com/valhalla/valhalla/actions/runs/6554539168/job/17801764898#step:5:5). Please use Material's 'material.extensions.emoji.twemoji'
as mkdocs_material_extensions is deprecated and will no longer be
supported moving forward. This is the last release.

  File "/home/runner/.local/lib/python3.10/site-packages/materialx/emoji.py", line 106, in twemoji
    return _patch_index(options)
  File "/home/runner/.local/lib/python3.10/site-packages/materialx/emoji.py", line [5](https://github.com/valhalla/valhalla/actions/runs/6554539168/job/17801764898#step:5:6)9, in _deprecated_func
    warnings.warn(

WARNING -  Material emoji logic has been officially moved into mkdocs-material
version 9.4. Please use Material's 'material.extensions.emoji.twemoji'
as mkdocs_material_extensions is deprecated and will no longer be
supported moving forward. This is the last release.

INFO    -  Cleaning site directory
INFO    -  Building documentation to directory: /home/runner/work/valhalla/valhalla/docs/site
INFO    -  The following pages exist in the docs directory, but are not included in the "nav" configuration:
  - api/turn-by-turn/add-routing-to-a-map.md
INFO    -  DeprecationWarning: 'materialx.emoji.to_svg' is deprecated.
Material emoji logic has been officially moved into mkdocs-material
version [9](https://github.com/valhalla/valhalla/actions/runs/6554539168/job/17801764898#step:5:10).4. Please use Material's 'material.extensions.emoji.to_svg'
as mkdocs_material_extensions is deprecated and will no longer be
supported moving forward. This is the last release.

  File "/home/runner/.local/lib/python3.[10](https://github.com/valhalla/valhalla/actions/runs/6554539168/job/17801764898#step:5:11)/site-packages/pymdownx/emoji.py", line 333, in handleMatch
    el = self.generator(
  File "/home/runner/.local/lib/python3.10/site-packages/materialx/emoji.py", line 59, in _deprecated_func
    warnings.warn(

WARNING -  Material emoji logic has been officially moved into mkdocs-material
version 9.4. Please use Material's 'material.extensions.emoji.to_svg'
as mkdocs_material_extensions is deprecated and will no longer be
supported moving forward. This is the last release.

INFO    -  Doc file 'index.md' contains an unrecognized relative link 'COPYING', it was left as is.
INFO    -  Doc file 'api/turn-by-turn/add-routing-to-a-map.md' contains an unrecognized relative link 'api-reference/#outputs-of-a-route', it was left as is. Did you mean 'api-reference.md#outputs-of-a-route'?
INFO    -  Doc file 'api/turn-by-turn/api-reference.md' contains an absolute link '/docs/decoding.md', it was left as is.

Aborted with 1 warnings in strict mode!
Error: Process completed with exit code 1.
```

--- a/.github/workflows/gh-pages.yml
+++ b/.github/workflows/gh-pages.yml
@@ -14,7 +14,7 @@ jobs:
 
     steps:
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
 
       - name: Set up Python 3
         uses: actions/setup-python@v4

--- a/docs/docs/api/turn-by-turn/add-routing-to-a-map.md
+++ b/docs/docs/api/turn-by-turn/add-routing-to-a-map.md
@@ -1,8 +1,0 @@
-# Add Turn-by-Turn routing to a map
-
-The [result of a routing request](api-reference/#outputs-of-a-route) is a special format that needs some processing to show in a JavaScript-based web map application.
-
-For [Leaflet](http://leafletjs.com/), the [Leaflet Routing Machine](https://www.liedman.net/leaflet-routing-machine/) via the [lrm-valhalla](https://github.com/valhalla/lrm-valhalla) plugin helps.
-Please note that [mapzen](https://github.com/mapzen/mapzen.js) urls are no longer available, so one must set the `serviceUrl` within the `options` of the `L.Routing.mapzen` router to point to a valid Valhalla route service!
-
-You can review the [documentation](api-reference.md) to learn more about routing with Turn-by-Turn.

--- a/docs/docs/api/turn-by-turn/api-reference.md
+++ b/docs/docs/api/turn-by-turn/api-reference.md
@@ -364,7 +364,7 @@ The summary JSON object includes:
 
 A `trip` contains one or more `legs`. For *n* number of `break` locations, there are *n-1* legs. `Through` locations do not create separate legs.
 
-Each leg of the trip includes a summary, which is comprised of the same information as a trip summary but applied to the single leg of the trip. It also includes a `shape`, which is an [encoded polyline](https://developers.google.com/maps/documentation/utilities/polylinealgorithm) of the route path (with 6 digits decimal precision), and a list of `maneuvers` as a JSON array. For more about decoding route shapes, see these [code examples](/docs/decoding.md).
+Each leg of the trip includes a summary, which is comprised of the same information as a trip summary but applied to the single leg of the trip. It also includes a `shape`, which is an [encoded polyline](https://developers.google.com/maps/documentation/utilities/polylinealgorithm) of the route path (with 6 digits decimal precision), and a list of `maneuvers` as a JSON array. For more about decoding route shapes, see these [code examples](../../decoding.md).
 
 If `elevation_interval` is specified, each leg of the trip will return `elevation` along the route as a JSON array. The `elevation_interval` is also returned. Units for both `elevation` and `elevation_interval` are either meters or feet based on the input units specified. 
 

--- a/docs/mkdocs.yml
+++ b/docs/mkdocs.yml
@@ -45,8 +45,8 @@ markdown_extensions:
   - pymdownx.tasklist:
       custom_checkbox: true
   - pymdownx.emoji:
-      emoji_index: !!python/name:materialx.emoji.twemoji
-      emoji_generator: !!python/name:materialx.emoji.to_svg
+      emoji_index: !!python/name:material.extensions.emoji.twemoji
+      emoji_generator: !!python/name:material.extensions.emoji.to_svg
 
 site_name: Valhalla Docs
 repo_url: https://github.com/valhalla/valhalla/

--- a/docs/requirements.txt
+++ b/docs/requirements.txt
@@ -1,1 +1,1 @@
-mkdocs-material>=8.5, <9.0
+mkdocs-material<10.0

